src/core/server/models/comment.ts

```diff
@@ -273,6 +303,96 @@ export async function retrieveCommentRepliesConnection(
   return retrieveConnection(input, query);
 }
 
+/**
+ * retrieveCommentParentsConnection will return a comment connection used to
+ * represent the parents of a given comment.
+ *
+ * @param mongo the database connection to use when retrieving comments
+ * @param tenantID the tenant id for where the comment exists
+ * @param commentID the id of the comment to retrieve parents of
+ * @param pagination pagination options to paginate the results
+ */
+export async function retrieveCommentParentsConnection(
+  mongo: Db,
+  tenantID: string,
+  comment: Comment,
+  { last: limit, before: skip = -1 }: { last: number; before?: number }
+): Promise<Readonly<Connection<Readonly<Comment>>>> {
+  // Return nothing if this comment does not have any parents.
+  if (!comment.parent_id) {
+    return createConnection({
+      pageInfo: {
+        hasNextPage: false,
+        hasPreviousPage: false,
+      },
+    });
+  }
+
+  // TODO: (wyattjoh) maybe throw an error when the limit is zero?
```

Comment:
I'd throw if all it does is returning an empty connection.

Reply:
An empty connection is still technically valid, albeit not super useful.